src/mono/mono/mini/intrinsics.c

```diff
@@ -1946,10 +1946,14 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 	if (in_corlib &&
 		!strcmp ("System.Runtime.CompilerServices", cmethod_klass_name_space) &&
 		!strcmp ("RuntimeFeature", cmethod_klass_name)) {
-		if (!strcmp (cmethod->name, "get_IsDynamicCodeSupported") || !strcmp (cmethod->name, "get_IsDynamicCodeCompiled")) {
+		if (!strcmp (cmethod->name, "get_IsDynamicCodeCompiled")) {
 			EMIT_NEW_ICONST (cfg, ins, cfg->full_aot ? 0 : 1);
```

Comment:
Isn't the result for `IsDynamicCodeCompiled` still wrong when the dynamic code is going to be interpreted?

Reply:
> Isn't the result for `IsDynamicCodeCompiled` still wrong when the dynamic code is going to be interpreted?

I don't think so.  It will be `IsDynamicCodeCompiled == false`.

The code in intrinsics.c is only used by the JIT and the AOT compiler.  In the interpreter, `IsDynamicCodeCompiled` is an intrinsic that returns false.  So any interpreted calls to it will return `false`, as expected.  In any code that is compiled ahead of time, it will also be false.

Reply:
> The code in intrinsics.c is only used by the JIT and the AOT compiler.

That's what I was missing. Thanks for the explanation.